decryptor/base/observers.go

```diff
@@ -61,11 +62,50 @@ func NewOnQueryObjectFromQuery(query string) OnQueryObject {
 	return &onQueryObject{query: query}
 }
 
-// QueryObserver will be used to notify about coming new query
+// BoundValue is a value provided for prepared statement execution.
+// Its exact type and meaning depends on the corresponding query.
+type BoundValue interface {
+	Data() []byte
+	Encoding() BoundValueEncoding
+}
+
+// BoundValueEncoding specifies how to interpret the bound data.
+type BoundValueEncoding int
+
+// Supported values of BoundValueEncoding.
+const (
+	BindText BoundValueEncoding = iota
+	BindBinary
+)
+
+type boundValue struct {
+	data     []byte
+	encoding BoundValueEncoding
```

Comment:
PostgreSQL calls these fields `format`, and not `encoding`. See [this link](https://www.postgresql.org/docs/8.2/protocol-message-formats.html). Consider renaming this field to `format`, as well as `BoundValueEncoding` -> `BoundValueFormat` and `Encoding()` -> `Format()` in `type BoundValue interface`

Reply:
> Consider renaming [encoding] field to `format`

Make sense. Renamed.